### PR TITLE
Excise sq_rsp

### PR DESCRIPTION
**Pull request by TiborGY**
## Description

It is done. The beast lays slain.
This PR replaces the last calls of the `sq_rsp` function in the `dfoccwave::Tensor2d::diagonalize` overloads with the new DSYEV wrappers, and removes the function and its file.

## User API & Changelog headlines

- [x] Deprecated `PSI_API` function `void sq_rsp(int nm, int n, double **array, double *evals, int matz, double **evecs, double toler)` has been removed.

## Dev notes & details

- [x] Merge `dfoccwave::Tensor2d::diagonalize` overloads to minimize code dupliaction
- [x] Add machine-readable docstrings in the header file
- [x] `dfoccwave::Tensor2d::diagonalize` overloads now throw on non-square matrices or diagonalizer failure
- [x] Replace calls of `sq_rsp` with `DSYEV_ascending` and `DSYEV_descending`
- [x] Delete `sq_rsp` function and file

## Checklist
- [x] No new features
- [x] CI tests are passing

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by TiborGY** [comment]
For some unknown reason ADCC tests run by the CI have started failing, but otherwise this is ready.

File: psi4/src/psi4/dfocc/tensors.h [hunk 1]
```diff
@@ -209,10 +209,24 @@ class Tensor2d {
...
+                     bool ascending = true);
+
+    /// @brief diagonalizes the Tensor2d using LAPACK DSYEV
+    /// @param dim number of rows (and columns)
+    /// @param eigvectors matrix of eigenvectors (2D row major array, one column for each eigvector)
+    /// @param eigvalues array to hold eigenvalues
+    /// @param cutoff defunct argument, setting it is a no-op, will be removed in the future
+    /// @param ascending Should the eigvalues be returned in ascending order?
+    void diagonalize(int dim, const SharedTensor2d &eigvectors, const SharedTensor1d &eigvalues, double cutoff,
+                     bool ascending);
```
Comment (davpoolechem) [inline psi4/src/psi4/dfocc/tensors.h:228]: Would it make sense to have `ascending` for this overload default to `true`, as well?

Comment (TiborGY) [reply psi4/src/psi4/dfocc/tensors.h:228]: Probably yes, but I wanted to keep the interfaces as untouched as reasonable to minimize dfocc disruption for the time being. They will need to be changed anyways after the rest of dfocc gets merged.

Comment (davpoolechem) [reply psi4/src/psi4/dfocc/tensors.h:228]: That's reasonable!

**Review by davpoolechem — APPROVED** [review]

**Review by JonathonMisiewicz — APPROVED** [review]
I'm happy with this. @loriab, will this cause an issue for the remaining dfocc resync?

**Comment by loriab** [comment]
I think it's ok. Here's the `diagonalize`s waiting to be used. https://github.com/psi4/psi4/pull/2684#issuecomment-1499733138 . There's some with the removed 3-arg call, but that has an implicit ascending, so they can be given the 4-arg version.

**Pull request merged**